src/main/scala/firrtl/passes/wiring/WiringTransform.scala

```diff
@@ -11,86 +11,87 @@ import scala.collection.mutable
-  def apply(target: ComponentName, pin: String): Annotation = Annotation(target, classOf[WiringTransform], s"source $pin")
+  def apply(target: ComponentName, pin: String): Annotation =
+    Annotation(target, classOf[WiringTransform], s"source $pin")
 
   private val matcher = "source (.+)".r
   def unapply(a: Annotation): Option[(ComponentName, String)] = a match {
-    case Annotation(ComponentName(n, m), _, matcher(pin)) => Some((ComponentName(n, m), pin))
+    case Annotation(ComponentName(n, m), _, matcher(pin)) =>
+      Some((ComponentName(n, m), pin))
     case _ => None
   }
 }
 
-/** A module, e.g. ExtModule etc., that should add the input pin
-  */
+/** An extractor of annotation sink components or modules */
 object SinkAnnotation {
-  def apply(target: ModuleName, pin: String): Annotation = Annotation(target, classOf[WiringTransform], s"sink $pin")
+  def apply(target: Named, pin: String): Annotation =
+    Annotation(target, classOf[WiringTransform], s"sink $pin")
 
   private val matcher = "sink (.+)".r
-  def unapply(a: Annotation): Option[(ModuleName, String)] = a match {
-    case Annotation(ModuleName(n, c), _, matcher(pin)) => Some((ModuleName(n, c), pin))
+  def unapply(a: Annotation): Option[(Named, String)] = a match {
+    case Annotation(ModuleName(n, c), _, matcher(pin)) =>
+      Some((ModuleName(n, c), pin))
+    case Annotation(ComponentName(n, m), _, matcher(pin)) =>
+      Some((ComponentName(n, m), pin))
     case _ => None
   }
 }
 
-/** A module under which all sink module must be declared, and there is only
-  * one source component
+/** Wires a Module's Source Component to one or more Sink
+  * Modules/Components
+  *
+  * Sinks are wired to their closest source through their lowest
+  * common ancestor (LCA). Verbosely, this modifies the circuit in
+  * the following ways:
+  *   - Adds a pin to each sink module
+  *   - Punches ports up from source signals to the LCA
+  *   - Punches ports down from LCAs to each sink module
+  *   - Wires sources up to LCA, sinks down from LCA, and across each LCA
+  *
+  * @throws WiringException if a sink is equidistant to two sources
   */
-object TopAnnotation {
-  def apply(target: ModuleName, pin: String): Annotation = Annotation(target, classOf[WiringTransform], s"top $pin")
+class WiringTransform extends Transform {
+  def inputForm: CircuitForm = MidForm
+  def outputForm: CircuitForm = MidForm
 
-  private val matcher = "top (.+)".r
-  def unapply(a: Annotation): Option[(ModuleName, String)] = a match {
-    case Annotation(ModuleName(n, c), _, matcher(pin)) => Some((ModuleName(n, c), pin))
-    case _ => None
-  }
-}
+  /** Defines the sequence of Transform that should be applied */
+  private def transforms(w: Seq[WiringInfo]): Seq[Transform] = Seq(
+    new Wiring(w),
+    ToWorkingIR,
+    InferTypes,
+    ExpandConnects,
+    Uniquify,
+    ExpandWhens,
+    CheckInitialization,
+    InferTypes,
+    ResolveKinds,
+    ResolveGenders
+  )
```

Comment:
Rather than including a mishmash of transforms that essentially perform HighForm -> MidForm should we instead just make this transformation return HighForm?

Reply:
Seems reasonable to me as this has ballooned from four Passes to all the passes... Just need to play musical chairs with the test output FIRRTL orderings...